src/fileformats/ocd_georef.h

```diff
@@ -0,0 +1,115 @@
+/*
+ *    Copyright 2018 Libor Pecháček
+ *
+ *    This file is part of OpenOrienteering.
+ *
+ *    OpenOrienteering is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    OpenOrienteering is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with OpenOrienteering.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OCD_GEOREF_H
+#define OCD_GEOREF_H
+
+#include <functional>
+#include <utility>
+#include <cmath>
+#include <limits>
+
+#include <QString>
+
+#include "core/georeferencing.h"
+
+namespace OpenOrienteering {
+
+class OcdGeoref
+{
+	Q_DECLARE_TR_FUNCTIONS(OpenOrienteering::OcdGeoref)
+
+public:
+	struct OcdStringFields
+	{
+		// Named after OCD string fields. Structure element name
+		// corresponds to the code and its value is the field value.
+		double a;
+		int m, x, y, i;
+
+		OcdStringFields() :
+		    a(std::nan("OCD")),
+		    m(-1),
+		    x(std::numeric_limits<int>::min()),
+		    y(std::numeric_limits<int>::min()),
+		    i(-1)
+		{ }
+	};
+
+	/**
+	 * @brief Fill in provided georef with data from type 1039 string.
+	 * @param georef Reference to a Georeferencing object.
+	 * @param param_struct Data extracted from OCD type 1039 string.
+	 * @param warning_handler Function to handle import warnings.
+	 */
+	static void georefFromFields(Georeferencing& georef,
+	                             const OcdStringFields& param_struct,
+	                             const std::function<void (const QString&)>& warning_handler);
```

Comment:
I recall that we discussed the traits of `Georeferencing` earlier but cannot find the reference. Anyway, my original proposal was passing `Georeferencing` by value but it is wrong because it inherits from `QObject`. We agreed that `Georeferencing` should not serve two purposes and should be freed from its `QObject` duties.

While writing this, I realized that for now the function can be called `setupGeorefFromFields()`. We can return to the original intent of passing by value after `Georeferencing` is fixed.